### ISSUE TITLE
Finetuning possible with images and associated semantic/instance segmented masks?

### ISSUE THREAD

**Issue by ajinkya-kulkarni** [computational-cell-analytics/micro-sam#400]
Hello team, thanks for this repo. 
I remember asking this a long time back, but maybe things have changed now. Is it possible to finetune the model(s) based on images and their corresponding masks? Similar to how we train a custom StarDist/Cellpose model.
It would be wonderful if this is possible!

Best, Ajinkya

**Comment by anwai98** [comment]
Hi @ajinkya-kulkarni,

Thanks for your interest. Yes, custom fine-tuning the Segment Anything models with `micro-sam` is possible using the images and their corresponding masks (which are used to automatically derive the relevant prompt inputs, e.g. points and box). There's been a recent contribution in `micro-sam` for automatic instance segmentation (which performs end-to-end instance segmentation). Here's an example script for [fine-tuning](https://github.com/computational-cell-analytics/micro-sam/blob/master/examples/finetuning/finetune_hela.py) SAM and here's the [briefing](https://github.com/computational-cell-analytics/micro-sam/tree/master/examples/finetuning) on how to make use of automatic instance segmentation.

**Comment by ajinkya-kulkarni** [comment]
Nice! I will try it out, thanks!
